plan/validator.go

```diff
@@ -302,3 +307,49 @@ func checkDuplicateColumnName(indexColNames []*ast.IndexColName) error {
 	}
 	return nil
 }
+
+// checkFieldLengthLimitation check the maximum length of the column
+// See https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html
+func checkFieldLengthLimitation(colDef *ast.ColumnDef) error {
+	tp := colDef.Tp
+	if tp == nil {
+		return nil
+	}
+	if tp.Flen > 4294967295 {
+		return errors.Errorf("Display width out of range for column '%s' (max = 4294967295)", colDef.Name.Name.O)
+	}
+	switch tp.Tp {
+	case mysql.TypeString:
+		if tp.Flen != types.UnspecifiedLength && tp.Flen > 255 {
+			return errors.Errorf("Column length too big for column '%s' (max = 255); use BLOB or TEXT instead", colDef.Name.Name.O)
+		}
+	case mysql.TypeVarchar:
+		maxFlen := 65535
+		cs := tp.Charset
+		// TODO: TableDefaultCharset-->DatabaseDefaultCharset-->SystemDefaultCharset.
+		// TODO: Change TableOption parser to parse collate.
+		// Reference https://github.com/pingcap/tidb/blob/b091e828cfa1d506b014345fb8337e424a4ab905/ddl/ddl_api.go#L185-L204
+		if len(tp.Charset) == 0 {
+			cs = mysql.UTF8Charset
```

Comment:
@tiancaiamao 
Why https://github.com/pingcap/tidb/blob/b091e828cfa1d506b014345fb8337e424a4ab905/ddl/ddl_api.go#L114-L114 is "utf8"?
```
drop table if exists t;
create table t (c varchar(32));
show full columns from t;
```
output:
```
+-------+-------------+-----------+------+------+---------+-------+---------------------------------+---------+
| Field | Type        | Collation | Null | Key  | Default | Extra | Privileges                      | Comment |
+-------+-------------+-----------+------+------+---------+-------+---------------------------------+---------+
| c     | varchar(32) | utf8_bin  | YES  |      | NULL    |       | select,insert,update,references |         |
+-------+-------------+-----------+------+------+---------+-------+---------------------------------+---------+
```
default should be mysql.UTF8Charset?